### ISSUE TITLE
Comparison files for reference

### ISSUE THREAD

**Issue by fortiko** [domints/CarpurideLogoGenerator#1]
Hello, and thank you for a wonderful project!

It just so happens that I had requested a boot logo last year when I got my 103W, so I can enclose:

* the original "asset" that I sent to carpuride (02-apple-1280x480.png), the 
* the Carpuride-generated isp_part.bin and its hex variant (orig_isp_part.bin.gz, orig_isp_part.bin.hex.gz, both as GZ because Github doesn't like neither bin nor hex files)
* the output of your generator isp_part.bin and its hex variant (isp_part.bin.gz, isp_part.bin.hex.gz, both as GZ because Github doesn't like neither bin nor hex files), and finally 
* a diff for both isp_part.bin files (rather, their hex variants) that shows a few small differences. 

Perhaps it is interesting to check or explain what causes those differences and then either adjust your code to 100% reproduce "Carpuride" files, or at least have this issue as part of the documentation why the files differ?

Great job in any case!

![02-apple-1280x480](https://github.com/user-attachments/assets/a4bc5dff-a3bd-47c2-a100-f6f4b289dc2e)
[orig_isp_part.bin.gz](https://github.com/user-attachments/files/16414502/orig_isp_part.bin.gz)
[orig_isp_part.bin.hex.gz](https://github.com/user-attachments/files/16414506/orig_isp_part.bin.hex.gz)
[isp_part.bin.gz](https://github.com/user-attachments/files/16414519/isp_part.bin.gz)
[isp_part.bin.hex.gz](https://github.com/user-attachments/files/16414559/isp_part.bin.hex.gz)
[diff.txt](https://github.com/user-attachments/files/16414632/diff.txt)

**Comment by domints** [comment]
Thanks for the feedback!
The difference is in colour data - I am pretty sure whatever they're using to generate bin files is for some reason recompressing the file using lossy compression algorithm. Maybe their tool for some convoluted reason only takes JPG, so they convert PNG (which is pixel perfect) to JPG which adds compression artifacts, and then they get the colour data to put into the bin file - which is more or less like converting it to BMP or PNG file again.

Here's image unpacked from the orig_isp_par.bin, and if you pixel-peep to compare the files you can see how edges are not as sharp anymore:
![logo bin](https://github.com/user-attachments/assets/f9c49adf-88e2-454d-9b99-6d70d425b7ab)

However I must admit it's barely visible on file like that. It was much more noticeable on the BMW logo my friend had sent them, BMW has this white/blue cross in the middle, source image was pixel perfect sharp, but logo he got from Carpuride was a bit fuzzy along the edge. Unfortunately I can't find the files to present it to you.

To summarize, it's just colour data, for some reason Carpuride doesn't generate bin file with exactly the same image you gave them, but it gets recompressed in the process. My generator is pixel perfect.


EDIT: As I think of it, you can convert both images (source one, you sent them, and the one from the bin, in my comment) to something like BMP, because it's very simple, straightforward format and do the same hex comparison - you'll notice that indeed image you got from the Carpu is not what you sent them :) You should get the same or similar diffs, except with colour byte order shift (and carpu has 4-byte colour, seemingly for useles alpha channel, always FF).

**Comment by fortiko** [comment]
That's great to know, thank you very much.

I'll then proceed and use your isp_par.bin to get maximum quality!

**Comment by fortiko** [comment]
Just for completeness sake, there also seems to be a shift by a few pixels to the right in the carpuride bin, compared to the original.
